app.coffee

```diff
@@ -5,34 +5,58 @@ favicon      = require 'static-favicon'
 logger       = require 'morgan'
 cookieParser = require 'cookie-parser'
 bodyParser   = require 'body-parser'
-
 passport     = require 'passport'
 
-routes       = require './routes/index'
-buckets      = require './routes/buckets'
 
-RedisStore =  require('connect-redis')(session);
+use_secure_settings = process.env.USE_SSL is '1'
+sessionOptions =
+    secret: process.env.EXPRESS_SESSION_SECRET or 'keyboard cat'
+    resave: false
+    saveUninitialized: false
+    name: 'protected_s3.sid'
+    proxy: use_secure_settings
+    cookie:
+        maxAge: 30 * 24 * 60 * 60 * 1000          # 30 days
+        secure: use_secure_settings
+        domain: if process.env.DOMAIN == 'localhost' then null else process.env.DOMAIN
+
+if process.env.USE_REDIS_SESSION is '1' and process.env.REDIS_URL
+    redisClient = require('redis-url').connect(process.env.REDIS_URL)
+    redisClient.on 'error', (err) ->
+        console.log "REDIS -> emit: ERROR", err
+    options = client: redisClient
+    RedisStore =  require('connect-redis')(session);
+    sessionOptions.store = new RedisStore(options)
 
-sessionOptions = { secret: process.env.EXPRESS_SESSION_SECRET or 'keyboard cat' }
+    process.on 'SIGTERM', ->
+        redisClient.quit()
+        redisClient.unref()
+        redisClient = null
+        return
 
-if process.env.USE_REDIS_SESSION is '1'
-  sessionOptions.store = new RedisStore()
 
 app = express()
 
+if use_secure_settings
+  app.set('trust proxy', 1)
+
 app.set('views', path.join(__dirname, 'views'))
 app.set('view engine', 'jade')
 
 app.use(favicon())
+app.use(require('stylus').middleware(path.join(__dirname, 'public')))
+app.use(express.static(path.join(__dirname, 'public')))
 app.use(logger('dev'))
```

Comment:
this is only about the amount of log information sent to Papertrail, if any Papertrail is used.